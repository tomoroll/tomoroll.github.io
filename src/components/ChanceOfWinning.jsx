import React from 'react';
import PropTypes from 'prop-types';
import ValueSlider from './ValueSlider';

function ChanceOfWinning({
  chances, min, max, updateChances,
}) {
  return (
    <div className="form-group">
      <b>Chance of winning</b>
      <ValueSlider value={chances} min={min} max={max} updateValue={updateChances} addonText="&nbsp;%&nbsp;" />
    </div>
  );
}
ChanceOfWinning.propTypes = {
  chances: PropTypes.number.isRequired,
  min: PropTypes.number,
  max: PropTypes.number,
  updateChances: PropTypes.func.isRequired,
};
ChanceOfWinning.defaultProps = {
  min: 0,
  max: 100,
};

export default ChanceOfWinning;
