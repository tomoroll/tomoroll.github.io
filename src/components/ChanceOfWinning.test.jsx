import React from 'react';
import renderer from 'react-test-renderer';
import ChanceOfWinning from './ChanceOfWinning';

it('renders correctly', () => {
  const tree = renderer
    .create(<ChanceOfWinning chances={50} updateChances={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
