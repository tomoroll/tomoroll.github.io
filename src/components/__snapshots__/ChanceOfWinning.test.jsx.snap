// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-group"
>
  <b>
    Chance of winning
  </b>
  <div
    className="row"
  >
    <div
      className="input-group col-sm-3 col-lg-2"
    >
      <input
        className="form-control"
        onChange={[Function]}
        type="number"
        value={50}
      />
      <div
        className="input-group-append"
      >
        <span
          className="input-group-text text-monospace"
        >
           % 
        </span>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="rc-slider-rail"
          style={Object {}}
        />
        <div
          className="rc-slider-track"
          style={
            Object {
              "left": "0%",
              "width": "50%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={50}
          className="rc-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          role="slider"
          style={
            Object {
              "left": "50%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
    </div>
  </div>
</div>
`;
